sequana/biomart.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+#
+#  This file is part of Sequana software
+#
+#  Copyright (c) 2016-2020 - Sequana Development Team
+#
+#  Distributed under the terms of the 3-clause BSD license.
+#  The full license is in the LICENSE file, distributed with this software.
+#
+#  website: https://github.com/sequana/sequana
+#  documentation: http://sequana.readthedocs.io
+#
+##############################################################################
+import time
+import re
+import os
+import io
+
+from sequana.lazy import pandas as pd
+
+import colorlog
+logger = colorlog.getLogger(__name__)
+
+
+__all__ = ["Mart"]
+
+
+# not tested. This is tested trough bioservics and takes a long time
+class Mart:  # pragma: no cover
+    """
+
+        conv = Mart(dataset="mmusculus_gene_ensembl")
+        # you could choose hsapiens_gene_ensembl for instance
+        df = conv.query()
+        df.set_index("ensembl_gene_id")
+        conv.save(df)
+
+    The file can now be loaded in e.g. :class:`~sequana.enrichment.kegg.KeggPathwayEnrichment`
+    as a mapper of the ensemble identifier to external names understood by Kegg.
+
+    """
+
+    def __init__(self, dataset, mart="ENSEMBL_MART_ENSEMBL"):
+        logger.info("Init Mart")
+        from bioservices import BioMart
```

Comment:
Je me demande s'il y a un réel gain de l'avoir ici plutôt qu'en import global

Reply:
No reason really. Except that it avoids to import all bioservices if not needed